tv_apps/smart-home/js/card_picker.js

```diff
@@ -114,8 +120,21 @@
         }
 
         var appButton = CardUtil.createCardButton(card, true);
-        that.gridView.appendChild(appButton);
-        that.appButtons.push(appButton);
+        var key = String(defaultOrder[card.group] || 5);
+        if (!(card instanceof Deck)) {
+          var lang = document.documentElement.lang;
+          var name = that._cardManager.resolveCardName(card, lang);
+          if (name.raw) {
+            key += name.raw;
+          } else if (name.args && name.args.number !== undefined) {
+            key += '0'.repeat(10 - name.args.number.length) + name.args.number;
+          } else if (name.id) {
+            key += name.id;
+          } else {
+            key += '_';
```

Comment:
Due to spec Jack updated to me a few days ago, we should sort only those cards in cardlist. (e.g. cards in folderList don't need to be sorted)

And by the current design, if we don't put `folder` cards together in front of the list, things would get wrong. Maybe this implication is not a good design so you can consider changing `every` to `forEach` here https://github.com/mozilla-b2g/gaia/pull/34241/files#diff-3d7aba178b4738a2cba1c8a8ab1f40e1R186